datafusion/core/tests/sql/decimal.rs

```diff
@@ -18,839 +18,256 @@
 use super::*;
 
 #[tokio::test]
-async fn decimal_cast() -> Result<()> {
+async fn decimal_cast_precision() -> Result<()> {
     let ctx = SessionContext::new();
     let sql = "select cast(1.23 as decimal(10,4))";
     let actual = execute_to_batches(&ctx, sql).await;
     assert_eq!(
```

Comment:
Like below, I think you can write these tests using `arrow_typeof` -- like 

```sql
❯ select arrow_typeof(cast(1.23 as decimal(10,4)));
+----------------------------+
| arrowtypeof(Float64(1.23)) |
+----------------------------+
| Decimal128(10, 4)          |
+----------------------------+
```

Reply:
I moved the tests to .slt file https://github.com/apache/arrow-datafusion/pull/5086/commits/8295aa5f7a138d1d44d3c05c436b88ef0b82140c